src/components/app.tsx

```diff
@@ -113,14 +113,16 @@ class App extends React.PureComponent<Props & {match: any; location: any; showEx
 
   public async setGist(parameter: {id: string; filename: string; revision?: string}) {
     try {
-      const response = await fetch(
+      const gistResponse = await fetch(
         `https://api.github.com/gists/${parameter.id}${
           parameter.revision !== undefined ? `/${parameter.revision}` : ''
         }`
       );
-      const data = await response.json();
-      const content = data.files[parameter.filename].content;
-      const contentObj = JSON.parse(content);
+      const gistData = await gistResponse.json();
```

Comment:
Does this mean we load small files multiple times? Can we avoid that?

Reply:
All gists are loaded multiple times if you use the dialogue
1. in `src/components/header/gist-modal/renderer.tsx` to check if it is valid
2. in `src/components/app.tsx` to actually use it

In `src/components/header/gist-modal/renderer.tsx` you could fall back to read from the raw url if the json is invalid, and thus potentially truncated - but that didn't feel quite right.

In `src/components/app.tsx` it should be fine to read directly from the raw url.

Reply:
Makes sense. Thanks for the explanation.

Reply:
I encountered one problem in `src/components/app.tsx`:
To directly read from the raw url I need the username, e.g.:
https://gist.githubusercontent.com/keckelt/83f0fdce4fcd5b758ba16e8570f76d92/raw/d1c9afb38a4129df7a6ef2963667c45032ba020b/Bin_Transform.vl.json

But the parameters there are taken from the URL (`https://vega.github.io/editor/#/gist/83f0fdce4fcd5b758ba16e8570f76d92/Bin_Transform.vl.json`) so it is not available there.

Reply:
Do you mean we could refactor to directly load the file if we knew the username? If so, can you file an issue about this?

Reply:
yes.
I assume you would need a second gist route like `https://vega.github.io/editor/#/gist/<username>/<gist-hash>/<filename>`
e.g.:
`https://vega.github.io/editor/#/gist/keckelt/83f0fdce4fcd5b758ba16e8570f76d92/Hist with null.vl.json`

Reply:
Ahh, no, we don't want that probably. I was thinking we could use the user name of the logged in user but that's only true for some gists.